### PR TITLE
Jdr gallery garnetted

### PR DESCRIPTION
**Pull request by Joshua-David**
## What does this change?

- Updates the header styles for both gallery and picture essay to be in line with garnett styles (same as immersive)
- Some changes to structure so they are both similar in layout, particularly around placement of standfirst/meta
- Removed the section label which is not needed on any immersive type
- Removed weird br tag in galleries which was causing alignment issues for captions throughout, handle spacing with css
- Tidied submeta 

Before - Web:
screen shot 2018-04-25 at 11 55 59

screen shot 2018-04-25 at 11 56 37

After - Web:
screen shot 2018-04-25 at 11 55 45

screen shot 2018-04-25 at 11 56 27

Before - Mobile:
screen shot 2018-04-25 at 11 57 41

After - Mobile:
screen shot 2018-04-25 at 17 21 32

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?
Styling for badges is included

Labs:
screen shot 2018-07-06 at 17 56 40

### CODE REVIEW

**Comment by paperboyo** [comment]
Thanks for the screenshots. That’s some serious black 😎!

More generally, and much less usefully, an observation: I think editors do not see fluid immersive picture slot for what it is – a forever-moving target. Instead, they tend to painstakingly go back-and-forth between Composer and preview and try to make the images “look good” (on their current monitor setup, and, hopefully, in Composer’s mobile preview).

These designs make current crops much shorter in portrait (by ~half in the above example) and slightly wider in landscape. And for a good reason too: otherwise the opaque black blob would cover even more of the image with its unpredictable, unpreviewable and on-headline-length-dependable dark nature.

It may be that the new ratio variation is always smaller than in the previous iteration (?).

Not sure what am I saying here...

**Comment by Joshua-David** [comment]
@paperboyo actually that wasn't intentional.... I've brought the old behaviour back and updated the screenshot (looks slightly different between old/new but it's the same in practice). There's no reason to change this behaviour

**Comment by paperboyo** [comment]
> There's no reason to change this behaviour

Until someone points out that now the crops are even more unpredictable because images are covered by the differing amounts of black. And this can be the whole bottom half of the image on mobile :-(

Completely unrelated: our `100vh` used to take into account the shown/hidden browser address bar, but I can’t find this working anywhere anymore...

**Comment by zeftilldeath** [comment]
Looks so nice. Does this share any of the markup from immersives?

**Comment by Joshua-David** [comment]
@paperboyo no worries. Overall they should be happier (I know I'm dreaming) with the new behaviour. You actually get less of the image covered, the headline and black background sit outside of the main media now and always start from the same place, so even if their headline is 100 words long it will only ever spill down into the main article itself 😎People will probably complain though because, people!

**Comment by NataliaLKB** [comment]
Have you checked with commercial dev about this change? Always good to keep them in the loop! This definitely affects some glabs pages

**Comment by Joshua-David** [comment]
@NataliaLKB good call. I've added labs to the screenshots to show how they are affected. Also looped in the labs team in chat and linked them here, seem happy with the changes

**Comment by Joshua-David** [comment]
@NataliaLKB @zeftilldeath @paperboyo @sndrs  anyone want to  👍 this? Be good to get it out

File: applications/app/views/fragments/galleryBody.scala.html [hunk 1]
```diff
@@ -18,13 +18,9 @@
...
         )"
         itemscope itemtype="@page.item.metadata.schemaType" role="main">
-
         @fragments.galleryHeader(page)
 
-        <div class="@RenderClasses(
-                    Map("l-side-margins--paidgallery" -> isPaidContent(page)),
-                    "l-side-margins", "l-side-margins--media", "l-side-margins--gallery"
-                )">
+        <div>
```
Comment (NataliaLKB) [inline applications/app/views/fragments/galleryBody.scala.html]: Is this div necessary now? Can it be deleted?

File: static/src/stylesheets/module/content-garnett/_gallery.head.scss [hunk 2]
```diff
@@ -1,65 +1,229 @@
...
-        margin-left: 0;
-        min-height: auto;
-        margin-bottom: $gs-baseline;
+.immersive-header-container {
+    background-color: $garnett-neutral-1;
+    overflow-x: hidden;
+}
+
+.immersive-header-container--photo-essay,
+.js-adblock-sticky {
```
Comment (NataliaLKB) [inline static/src/stylesheets/module/content-garnett/_gallery.head.scss]: `.js` classes shouldn't be used for styling. They are only for use in javaScript.

**Comment by Joshua-David** [comment]
@NataliaLKB should have addressed all the changes now

**Comment by Joshua-David** [comment]
@SiAdcock @sndrs are either of you able to do a review of this? It's been sitting for several days now with no love, would be great to get this out... 🙏

File: static/src/stylesheets/module/content-garnett/_gallery.head.scss [hunk 3]
```diff
@@ -1,65 +1,243 @@
...
+    overflow-x: hidden;
+}
+
+.immersive-header-container--photo-essay {
+    background-color: $garnett-neutral-5;
+}
+
+// Header and tags
+.immersive-main-media__gallery {
+    height: 100vh;
```
[the remark this answers is not in the document]
Comment (Joshua-David) [reply static/src/stylesheets/module/content-garnett/_gallery.head.scss]: will add, do we have a standard for adding fall backs in cases like this and the one below?

File: static/src/stylesheets/module/content-garnett/_gallery.scss [hunk 4]
```diff
@@ -9,21 +8,17 @@
...
 }
 
 .content__main-column--gallery {
     max-width: none;
     margin-right: auto;
-    padding-bottom: $gs-baseline*2;
 
     @include mq(desktop) {
         margin-left: gs-span(3) + $gs-gutter;
-        padding-left: 0 !important;
```
[the remark this answers is not in the document]
Comment (Joshua-David) [reply static/src/stylesheets/module/content-garnett/_gallery.scss]: it's being removed not added 😎

File: static/src/stylesheets/module/content-garnett/_gallery.head.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (NataliaLKB) [reply static/src/stylesheets/module/content-garnett/_gallery.head.scss]: Fallbacks have to by analysed on a case by case basis usually. You can use `@supports` in css to define when to use 100vh vs the fallback.

Documentation here: https://developer.mozilla.org/en-US/docs/Web/CSS/%40supports

File: static/src/stylesheets/module/content-garnett/_gallery.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (NataliaLKB) [reply static/src/stylesheets/module/content-garnett/_gallery.scss]: 💯

File: static/src/stylesheets/module/content-garnett/_gallery.head.scss [hunk 5]
```diff
@@ -1,65 +1,243 @@
...
     }
-}
 
-// Overriding the default for media tones
-.tonal--tone-media .content__headline--gallery {
-    font-weight: 200;
-    max-width: gs-span(8);
-    padding-top: 0;
+    .byline,
+    .byline span {
```
Comment (SiAdcock) [inline static/src/stylesheets/module/content-garnett/_gallery.head.scss]: Can we use the direct descendant selector here, for efficiency?

```css
.byline > span {
```

File: static/src/stylesheets/module/content-garnett/_gallery.scss [hunk 6]
```diff
@@ -271,3 +283,50 @@
         opacity: 1;
     }
 }
+
+#more-galleries {
```
Comment (SiAdcock) [inline static/src/stylesheets/module/content-garnett/_gallery.scss]: What is this targeting? I can't see any instances of this ID in our codebase

**Comment by Joshua-David** [comment]
@NataliaLKB @zeftilldeath @SiAdcock @sndrs, I bet you thought this PR was dead... 

I've addressed your previous comments, would you mind taking a look?

**Review by zeftilldeath — APPROVED** [review]
👍 As far as I can see it looks great. Excited to see it in the wild.

**Pull request merged**